src/addrman.cpp

```diff
@@ -394,8 +398,9 @@ CAddrInfo CAddrMan::Select_(bool newOnly)
                 nUBucketPos = (nUBucketPos + insecure_rand.randbits(ADDRMAN_BUCKET_SIZE_LOG2)) % ADDRMAN_BUCKET_SIZE;
             }
             int nId = vvNew[nUBucket][nUBucketPos];
-            assert(mapInfo.count(nId) == 1);
-            CAddrInfo& info = mapInfo[nId];
+            const auto it_found{mapInfo.find(nId)};
```

Comment:
Here and line 383 above, apparently using `auto` with `{}` can lead to surprising results and it's recommended to use `=` rather than `{}` for objects specified with `auto` when we don't mean a list (at least, for C++11 per my Stroustrup book, not sure if it changed since in 14 or 17, on first skim of https://en.cppreference.com/w/cpp/language/auto I didn't see a change).

Reply:
Mind to share an example to illustrate the "surprise"?

Reply:
![image](https://user-images.githubusercontent.com/2415484/118260513-f3ec0e00-b4a1-11eb-95e7-8c1423580760.jpeg)

![image](https://user-images.githubusercontent.com/2415484/118260555-00706680-b4a2-11eb-8a95-6f8d85e52d9d.jpeg)

(afk, sent from phone)

Reply:
See https://en.cppreference.com/w/cpp/language/template_argument_deduction#auto_type_deduction

Also, if this was a list, the code wouldn't compile.

Reply:
Thanks for the link. Seems it changed from 2014: https://isocpp.org/blog/2014/03/n3922, IIUC.

Reply:
"For direct list-initialization:

    For a braced-init-list with only a single element, auto deduction will deduce from that entry;
    For a braced-init-list with more than one element, auto deduction will be ill-formed."